### ISSUE TITLE
Possible improvements on Today's Task and Map

### ISSUE THREAD

**Issue by Pozolo** [chiyadev/genshin-schedule#18]
Maybe this has more to do on how I use the site but, I have my own routes marked on the map, currently only crystals and respawnable artifacts, and is quite tedious having to click on each icon twice (one for selecting it and one to mark as done) for every icon.
My suggestion is being able to have a parent task and its children, and checking the parent would mark all as done.

How the map could look:
![map-icons-and-routes](https://user-images.githubusercontent.com/75450558/101087891-2ddfec80-35b3-11eb-8779-1ef317ef8614.jpg)

How the list could look:
![task-list-con-child-tasks](https://user-images.githubusercontent.com/75450558/101087903-30424680-35b3-11eb-9e07-f716a680d8c9.jpg)

I've seen the post "Keyboard shortcuts for common actions" but that leads me to this issue:
About the task list, i would find it better if the task got arranged alphabetically, that way we could have more control with the order they appear. I'm not sure how they are currently arranged.

The last thing that i found missing is the possibility of having filters to pick specific tasks and avoid having the list and the map cluttered. Would be great being capable of having multiple swappable lists like: Ore location tasks, elite enemies locations, artifact locations, etc. I can solve this switching between different users on Google Chrome so I'm not too worried about this.

Also want to say that this is one of the best pages for Genshin Impact. Thanks for your hard work.

**Comment by luaneko** [comment]
> Maybe this has more to do on how I use the site but, I have my own routes marked on the map, currently only crystals and respawnable artifacts, and is quite tedious having to click on each icon twice

I agree with this. I was actually having the same problem because I added the daily artifact route on the website but wasn't sure how to approach it. I think this is a good suggestion.

> About the task list, i would find it better if the task got arranged alphabetically

The task list is currently sorted in the same order that appears when selecting the task icons, and for the same icons they are then sorted by due time, so that more overdue tasks come first in the list.

I'm not sure if alphabetical ordering would be a good idea.

> The last thing that i found missing is the possibility of having filters to pick specific tasks and avoid having the list and the map cluttered.

I was thinking of a search box above the list, but a dropdown of "task groups" can work too.

**Comment by luaneko** [comment]
Keyboard shortcuts have been implemented so it should somewhat mitigate this issue for now. You can press `n` to focus the next due task, and `d` to mark the focused task as done. Refer to the help window by pressing `k`.

**Comment by Pozolo** [comment]
For some reason I never noticed that the task were arranged like that. Since I had my artifacts marked as purple stones and some mora spots marked with yellow stones it confused me how they showed up on the list, but now I will change them to the same icon and make use of the shortcuts. Thanks.

**Comment by luaneko** [comment]
Sorry, it turns out I was actually misinformed. Tasks *are* currently [ordered by their icon name alphabetically](https://github.com/chiyadev/genshin-schedule/blob/ddc958b017725b75f371ba19dfadede64f5fd5f5/web/utils/tasks.ts#L15), and for the tasks with the same icon they are then ordered by due time.

I think fixing this to follow the same order as it appears in the icon list would be better.
